scripts/exit.py

```diff
@@ -4,19 +4,435 @@
+        deployment_addrs["RootChain"],
+    )
+    root_chain = Contract.from_explorer(root_chain_proxy_addr, root_chain_addr, silent=True)
+
+    start = 1
+    end = root_chain.currentHeaderBlock() // CHECKPOINT_ID_INTERVAL
+
+    header_block_number = None
+    while start <= end:
+        if start == end:
+            header_block_number = start
+
+        middle = (start + end) // 2
+        header_block = root_chain.headerBlocks(middle * CHECKPOINT_ID_INTERVAL)
+        header_start = header_block["start"]
+        header_end = header_block["end"]
+
+        if header_start <= child_block_number <= header_end:
+            header_block_number = middle
+            break
+        elif header_start > child_block_number:
+            # child block was checkpointed after
+            end = middle - 1
+        elif header_end < child_block_number:
+            # child block was checkpointed before
+            start = middle + 1
+
+    return header_start, header_end, header_block_number * CHECKPOINT_ID_INTERVAL
+
+
+def prepare_receipt(receipt: TxReceipt) -> PreparedReceipt:
+    """Prepare a transaction receipt for serialization"""
+    receipt_root = HexBytes(receipt.get("root", b""))
+    receipt_status = receipt.get("status", 1)
+
+    receipt_root_or_status = receipt_root if len(receipt_root) > 0 else receipt_status
+    receipt_cumulative_gas = receipt["cumulativeGasUsed"]
+    receipt_logs_bloom = HexBytes(receipt["logsBloom"])
+    receipt_logs = [
+        (
+            HexBytes(log["address"]),
+            list(map(HexBytes, log["topics"])),
+            HexBytes(log["data"]),
+        )
+        for log in receipt["logs"]
+    ]
+
+    return (
+        receipt_root_or_status,
+        receipt_cumulative_gas,
+        receipt_logs_bloom,
+        receipt_logs,
+    )
+
+
+def serialize_receipt(receipt: TxReceipt) -> bytes:
+    """Serialize a receipt.
+
+    This also handles EIP-2718 typed transactions.
+    """
+    prepared_receipt = prepare_receipt(receipt)
+    encoded_receipt = rlp.encode(prepared_receipt)
+
+    receipt_type = HexBytes(receipt.get("type", 0))
+    if receipt_type == HexBytes(0):
+        return encoded_receipt
+
+    buffer = HexBytes(receipt_type) + encoded_receipt
+    return rlp.encode(buffer)
+
+
+def serialize_block(block: dict) -> bytes:
+    """Serialize a block."""
+    block_number = block["number"].to_bytes(32, "big")
+    timestamp = block["timestamp"].to_bytes(32, "big")
+    txs_root = HexBytes(block["transactionsRoot"])
+    receipts_root = HexBytes(block["receiptsRoot"])
+    return keccak256(block_number + timestamp + txs_root + receipts_root)
+
+
+@hot_swap_network("polygon-main")
+def build_block_proof(block_start: int, block_end: int, burn_tx_block_number: int) -> List[bytes]:
+    """Build a merkle proof for the burn tx block."""
+    checkpoint_blocks = (
+        web3.eth.get_block(block_number) for block_number in range(block_start, block_end + 1)
+    )
```

Comment:
1. You could use brownie's `chain` object here
2. This is slooow and at first I thought the script was hanging. Some console output to appease impatient users would be pretty lovely.